### PR TITLE
circuits: Re-do `TxCircuit` construction

### LINKED ISSUE #229: circuits: Restructure API for circuit creation
#### Summary
Hide or even remove the `TxInputNote` and `TxOutputNote` structs to be able to create a circuit from only it's parameters. This is to improve the creation of a phoenix-transaction in `execution-core` without the need to depend on `phoenix-circuits` for it's types.

See [rusk #2069](https://github.com/dusk-network/rusk/issues/2069)

### PR DESCRIPTION
**Pull request by moCello**
Resolves #229

### CODE REVIEW

File: circuits/src/lib.rs [hunk 1]
```diff
@@ -13,8 +13,374 @@
...
+/// transaction output-notes.
+#[derive(Debug, Clone)]
+pub struct OutputNoteInfo {
+    /// The value of the note
+    pub value: u64,
+    /// The value-commitment of the note
+    pub value_commitment: JubJubAffine,
+    /// The blinder used to calculate the value commitment
+    pub value_blinder: JubJubScalar,
+    /// the public key of the note
```
Comment (xevisalle) [inline circuits/src/lib.rs:380]: T!! :)

File: circuits/src/lib.rs [hunk 2]
```diff
@@ -13,8 +13,374 @@
...
+            signatures,
+        }
+    }
+}
+
+/// Struct holding all information needed by the transfer circuit regarding the
+/// transaction input-notes.
+#[derive(Debug, Clone)]
+pub struct InputNoteInfo<const H: usize> {
+    /// the merkle opening for the note
```
Comment (xevisalle) [inline circuits/src/lib.rs:305]: Ts here, or make them lower case in the other struct

**Review by xevisalle — APPROVED** [review]
LGTM

File: circuits/src/lib.rs (hunk 1 above)
Comment (moCello) [reply circuits/src/lib.rs:380]: will fix in #233

File: circuits/src/lib.rs (hunk 2 above)
Comment (moCello) [reply circuits/src/lib.rs:305]: will fix in #233

**Pull request merged**